### PR TITLE
Fix building with Windows. #1684

### PR DESCRIPTION
**Pull request by gam2046**
Signed-off-by: forDream <gam2046@users.noreply.github.com>

### Type of changes

_Put an `x` inside the [ ] that applies._

* [x] Bugfix
* [ ] New feature
* [ ] Translation
* [ ] General refinement
* Other:

### Details

- Supports building projects on windows.
- Python is required

### CODE REVIEW

File: core/src/overture/make.bat [hunk 1]
```diff
@@ -28,21 +28,35 @@ MKDIR %TARGET%\armeabi-v7a>nul 2>nul
 MKDIR %TARGET%\x86>nul 2>nul 
 MKDIR %TARGET%\arm64-v8a>nul 2>nul 
 
+SET CC=%ANDROID_NDK_HOME%\toolchains\arm-linux-androideabi-4.9\prebuilt\windows-x86_64\bin\arm-linux-androideabi-gcc.exe
```
Comment (wongsyrone) [inline core/src/overture/make.bat]: It seems you are using the outdated ndk

Comment (gam2046) [reply core/src/overture/make.bat]: In fact, the ndk I used is not outdated.
But I found that maybe could be written like this:
`SET CC=%ANDROID_ARM_TOOLCHAIN%\bin\arm-linux-androideabi-gcc.exe`

Comment (madeye) [reply core/src/overture/make.bat]: I think this one is better. Please update the path here.

**Comment by wongsyrone** [comment]
CC variable is set below, why we need this at the very beginning?

**Comment by wongsyrone** [comment]
If you want to check the availability of `ANDROID_{ARCH}_CC`, I don't think this makes any sense, the toolchain installer makes sure it does exist.

And it seems  `WHERE python.exe` doesn't work on my box.

**Comment by gam2046** [comment]
`goBuild' requires gcc to compile(I'm not good at Golang, maybe gcc is needed in Golang), and there are no gcc environment variables by default for windows, so `CC` variables need to be set. So that the Golang compiler knows the location of the gcc.

I am sure that the `where` commands is available on Windows 7/10, and I compile it on Windows 10

**Comment by wongsyrone** [comment]
New NDK deprecates GCC and CC is set at https://github.com/gam2046/shadowsocks-android/blob/207d76a143a4e9bb9b0e1a58a7f122060aae63c2/core/src/overture/make.bat#L95

**Comment by gam2046** [comment]
https://github.com/shadowsocks/shadowsocks-android/blob/master/core/src/overture/make.bat#L53
https://github.com/shadowsocks/go/blob/master/src/make.bat#L29

Maybe gcc needs to be used in golang?

Now you define `CC` on line 95, and you need to use `CC` on line 53

**Comment by wongsyrone** [comment]
It's host compiler, not the cross-compiler. Since widely used Linux distro has gcc installed by default. That env won't cause problems. For Windows, you have to choose a host compiler to compile *.c src correctly. As far as I know, Go's bootstrap doesn't rely on C now.

**Comment by gam2046** [comment]
Is this related to the version of GO? 
At compile time, I use go 1.7 to indicate that gcc cannot be found.
Finally, I added gcc to allow the compilation to continue.

**Comment by wongsyrone** [comment]
Not sure about that, I don't use windows to compile this project. BTW, you want to use https://github.com/shadowsocks/go/tree/shadowsocks-go1.10, it's golang 1.10.

**Comment by gam2046** [comment]
emmmm I'm not good at `GO`. I don't know the relationship between these versions :
- Version of https://github.com/shadowsocks/go/
- Version of Go binary : https://golang.org/dl/
- Version of Go's bootstrap

I tried to compile this project in Linux, no problem.  But try to compile in Windows, encounter some problems, and try to fix these problems, one of the problems is that can't find gcc

**Comment by Mygod** [comment]
No additional comments from me. CC @imReker

**Comment by imReker** [comment]
I create windows build.bat in less than 5 minutes, just simply translate sh to bat, so it doesn't have error check, same as sh version. It just do right thing when everything is OK.

Actually the build.bat can't work on most people's machine because of the Build of GoLang, like #1575.
GCC for windows is required, and my solution is a Windows 10 machine and build GoLang within WSL, then build.bat will success. 

@wongsyrone 
`WHERE python.exe` will not work if python is not in %PATH%.
This check is OK because user will see an error message in console instead of a Open With window when execute py file.

**Comment by wongsyrone** [comment]
Then my opinion is building on windows doesn't worth fixing, there are so many issues ordinary users cannot handle. Just let who wants to compile use *nix.

**Comment by imReker** [comment]
Agree. Waste time on Windows build is not worth. I just make Windows build possible on some machine, but not perfect on every machine. If someone really want to build on Windows, read build.bat and related pr comment is enough.
Most people just need overture.so bin to finish the build progress.

So, add windows build.bat to `task goBuild` maybe is a bad idea. Remove it and people will think it can not be built on Windows.

**Pull request merged**